src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -35,7 +35,7 @@
     public Socket(EventBus eventBus, SocketHint<T> socketHint, T value) {
         this.eventBus = eventBus;
         this.socketHint = socketHint;
-        this.value = Optional.of(value);
+        this.value = value;
 
         checkNotNull(eventBus);
         checkNotNull(socketHint);
```

Comment:
Below this line you have `checkNotNull(value);` this can't be there. The value can be null by default.